testing/infra/terraform/modules/soaktest_workers/main.tf

```
@@ -0,0 +1,141 @@
+locals {
+  executable_name    = "apmsoak"
+  service_name       = "apmsoak.service"
+  network_name       = "apmsoak-network"
+  remote_working_dir = "apmsoak/"
+  config_hash = sha256(join("_",
+    [
+      var.apm_server_url,
+      var.apm_secret_token,
+      var.apm_loadgen_max_rate,
+      var.apm_loadgen_agents_replicas,
+      var.elastic_agent_version,
+      var.fleet_url,
+    ]
+  ))
+}
+
+data "google_compute_image" "worker_image" {
+  project = "ubuntu-os-cloud"
+  family  = "ubuntu-minimal-2004-lts"
+}
+
+data "google_client_openid_userinfo" "me" {}
+
+resource "google_os_login_ssh_public_key" "cache" {
+  user = data.google_client_openid_userinfo.me.email
+  key  = file(var.public_key)
+}
+
+resource "google_compute_network" "apmsoak_worker_network" {
+  name = local.network_name
+}
+
+resource "google_service_account" "worker_svc_account" {
+  account_id   = "apmserver-soaktest-worker"
+  display_name = "APM Server soaktest worker"
+}
+
+resource "google_compute_firewall" "allow_ssh" {
+  name    = "apmserver-soaktest-allowssh"
+  network = local.network_name
+
+  depends_on = [
+    google_compute_network.apmsoak_worker_network
+  ]
```

We can remove this `depends_on` block if we use the `google_compute_network.apmsoak_worker_network.self_link`. See https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_firewall#network and https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_network#self_link.

```suggestion
  network = google_compute_network.apmsoak_worker_network.self_link
```

---

Thanks, sounds cleaner

---

Fixed